django_project/ford3/views/user_management.py

```diff
@@ -22,23 +22,27 @@
 
 def show(request):
     # prepare to list user management
-    group_id = request.user.groups.all().first().id
-    if group_id == 1:
-        user_mgmt = UserManagement.provider_objects.all()
-    elif group_id == 3:
-        user_mgmt = UserManagement.campus_objects.all()
-    elif group_id == 2:
-        return custom_403(request)
-    # superuser
-    else:
+    # for admin
+    if request.user.is_staff:
         user_mgmt = UserManagement.objects.all()
-    # prepare form for add new user
-    group_name = request.user.groups.all().first().name
-    allowed_group = ''
-    if group_name == 'Provinces':
-        allowed_group = 'Providers'
-    elif group_name == 'Providers':
-        allowed_group = 'Campus'
+        # admin can add a user belongs to Provinces
+        allowed_group = 'Provinces'
+    else:
+        # for non-admins
+        group_id = request.user.groups.all().first().id
+        if group_id == 1:
+            user_mgmt = UserManagement.provider_objects.all()
+        elif group_id == 3:
+            user_mgmt = UserManagement.campus_objects.all()
+        elif group_id == 2:
+            return custom_403(request)
+        # prepare form for add new user
+        group_name = request.user.groups.all().first().name
+        allowed_group = ''
+        if group_name == 'Provinces':
```

Comment:
what if we change the name of the group later in the admin? (As we already did in production)

Reply:
yes, i though about this as well.
i'm thinking to stick with the `id` instead of `name`. will the id change too?

Reply:
likely not